src/semaphore_internal.h

```diff
@@ -29,45 +29,80 @@
 
 struct dispatch_queue_s;
 
-#define DISPATCH_SEMAPHORE_HEADER(cls, ns) \
-	DISPATCH_OBJECT_HEADER(cls); \
-	intptr_t volatile ns##_value; \
-	_dispatch_sema4_t ns##_sema
-
-struct dispatch_semaphore_header_s {
-	DISPATCH_SEMAPHORE_HEADER(semaphore, dsema);
-};
-
-DISPATCH_CLASS_DECL(semaphore);
+DISPATCH_CLASS_DECL(semaphore, OBJECT);
 struct dispatch_semaphore_s {
-	DISPATCH_SEMAPHORE_HEADER(semaphore, dsema);
+	DISPATCH_OBJECT_HEADER(semaphore);
+	long volatile dsema_value;
```

Comment:
Is this supposed to be 32-bit always or should this be `intptr_t` so that it is 64-bit on 64-bit targets and 32-bit on 32-bit targets?

Reply:
To match the behavior elsewhere, this should be intptr_t.